### PR TITLE
Updated year on hkdf.c

### PR DESCRIPTION
**Pull request by cm3173**
Updated the year on the license disclaimer. @bbbrumley 

CLA: trivial

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: demos/kdf/hkdf.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021-2023 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2021-2024 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (tom-cosgrove-arm) [inline demos/kdf/hkdf.c:2]: Hmm, it doesn't look to me that this file was updated in 2024 - the last change I see in here is

```
commit 86db958835d1f8ba9ce49a9f93b5309c3d13b91c
Author: James Muir <james@openssl.org>
Date:   Fri Nov 10 14:02:00 2023 -0500

    demos: tidy up makefiles, fix warnings
```

**Comment by mattcaswell** [comment]
Generally speaking updating the copyright year in files like this is not normally necessary. We have a script that does this when we create a release from a branch (i.e. if a year is wrong it will be corrected when the contents of the master branch become the 3.4 release at some point in the future).

https://github.com/openssl/tools/blob/master/release-tools/do-copyright-year

**Comment by bbbrumley** [comment]
Thanks both for the feedback

@cm3173 In light of the comments, probably best to close this one out

Nevertheless, thank you for submitting your PR

**Comment by cm3173** [comment]
I will be closing it now. Thank you for working on this great project :)

**Pull request closed without merging**